packages/amplify-cli/src/commands/gen2-migration/refactor/resolvers/cfn-dependency-resolver.ts

```diff
@@ -0,0 +1,29 @@
+import { CFNTemplate } from '../types';
+
+class CfnDependencyResolver {
+  constructor(private readonly template: CFNTemplate) {}
+
+  public resolve(resourcesToRefactor: string[]) {
+    const clonedGen1Template = JSON.parse(JSON.stringify(this.template)) as CFNTemplate;
+    const resources = clonedGen1Template.Resources;
+    Object.entries(resources).forEach(([logicalResourceId, resource]) => {
+      if (resource?.DependsOn) {
+        // Normalize DependsOn to array for processing
+        const deps = Array.isArray(resource.DependsOn) ? resource.DependsOn : [resource.DependsOn];
+        const depsInRefactor = deps.filter((dep: string) => resourcesToRefactor.includes(dep));
```

Comment:
Functional change here. Normalize [DependsOn](https://docs.aws.amazon.com/AWSCloudFormation/latest/TemplateReference/aws-attribute-dependson.html), which can be either a string or an array of strings, to be an array.